### PR TITLE
feat(wolfi): add support for Wolfi Linux

### PR DESCRIPTION
**Pull request by luhring**
This PR allows the Trivy vuln-list to incorporate data from Wolfi's secdb. (I'm on the Wolfi team 👋 )

**For some context:** Wolfi's secdb is very similar to Alpine's secdb, so much of the code is based on the existing Alpine-focused implementation. However, there's a few notable differences, such as that Wolfi doesn't have version streams like Alpine does (e.g. 3.15, 3.16, edge, etc.). It's also not guaranteed that Wolfi's secdb will forever remain the shape of the Alpine secdb since these are two separate and distinct projects.

Related to https://github.com/aquasecurity/trivy/issues/3205

### CODE REVIEW

**Review by knqyf263 — COMMENTED** [review]
Thanks for your contribution. I left a comment.

File: wolfi/wolfi.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+	log.Println("Fetching Wolfi data...")
+	if err = u.save(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (u Updater) traverse(url url.URL) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (luhring) [reply wolfi/wolfi.go]: Good catch! Thanks. Removed in 9709c1c.

**Comment by afdesk** [comment]
I've merged the alpine using.
Unfortunately, I couldn't create a new PR to this branch, so this one was updated

**Comment by afdesk** [comment]
@luhring @knqyf263 Could you take a look at these updates?
Now I'm not sure in this way, my main concern is the next message:

>  we have recently discussed a few enhancements to the data structure.

from here: https://github.com/aquasecurity/trivy/issues/3205#issuecomment-1325558829

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
Thank you guys. We can split into Alpine and Wolfi after Wolfi adopts a new data structure.

**Comment by luhring** [comment]
Thanks for the help @afdesk and @knqyf263! 🎉

**Comment by afdesk** [comment]
> Thanks for the help @afdesk and @knqyf263! tada

thank you for contribution. I hope it's ok, that i change your code

**Comment by luhring** [comment]
> thank you for contribution. I hope it's ok, that i change your code

Of course! It's important that you all are happy with the state of the code.

**Pull request merged**